### PR TITLE
Industry Autocomplete

### PR DESCRIPTION
**Pull request by amirulraziqi**
https://github.com/AdvisorySG/mentorship-page/issues/335

Industry autocomplete should work successfully.

### CODE REVIEW

File: src/components/search-bar.jsx [hunk 1]
```diff
@@ -67,13 +69,98 @@ const SearchBar = ({ mentors, setHasSearch, setSearchResults }) => {
...
+    () =>
+      [
+        ...new Set(
+          Object.values(mentors).flatMap(({ industries }) => industries)
+        ),
+      ].sort(),
+    [mentors]
+  );
+
+  useEffect(() => {
```
Comment (wei2912) [inline src/components/search-bar.jsx:82]: Please remove L82-84.

File: src/components/search-bar.jsx [hunk 2]
```diff
@@ -67,13 +69,98 @@ const SearchBar = ({ mentors, setHasSearch, setSearchResults }) => {
...
   return (
     <form noValidate autoComplete="off" className="search-bar">
-      <TextField
-        id="standard-search"
-        label="Search mentors by..."
-        onChange={(newValue) => setSearchValue(newValue.target.value)}
-      />
+      {field === "industry" ? (
+        <Downshift
+          id="downshift-multiple"
```
Comment (wei2912) [inline src/components/search-bar.jsx:92]: Please delete `id`?? if you can

File: src/components/search-bar.jsx [hunk 3]
```diff
@@ -67,13 +69,98 @@ const SearchBar = ({ mentors, setHasSearch, setSearchResults }) => {
...
+            getInputProps,
+            getItemProps,
+            isOpen,
+            inputValue,
+            selectedItem,
+            highlightedIndex,
+          }) => (
+            <div className={classes.container}>
+              <TextField
+                id="standard-search"
```
Comment (wei2912) [inline src/components/search-bar.jsx:108]: Please remove `id`

File: src/components/search-bar.jsx [hunk 4]
```diff
@@ -67,13 +69,98 @@ const SearchBar = ({ mentors, setHasSearch, setSearchResults }) => {
...
+                      );
+                    })}
+                </div>
+              ) : null}
+            </div>
+          )}
+        </Downshift>
+      ) : (
+        <TextField
+          id="standard-search"
```
Comment (wei2912) [inline src/components/search-bar.jsx:159]: Please remove

File: src/components/search-bar.jsx [hunk 5]
```diff
@@ -67,13 +69,98 @@ const SearchBar = ({ mentors, setHasSearch, setSearchResults }) => {
...
+            inputValue,
+            selectedItem,
+            highlightedIndex,
+          }) => (
+            <div className={classes.container}>
+              <TextField
+                id="standard-search"
+                label="Search mentors by..."
+                InputProps={{
+                  ...getInputProps(),
```
Comment (wei2912) [inline src/components/search-bar.jsx:111]: why not `InputProps={getInputProps()}`?

File: src/components/search-bar.jsx [hunk 6]
```diff
@@ -67,13 +69,98 @@ const SearchBar = ({ mentors, setHasSearch, setSearchResults }) => {
...
+                }}
+                onChange={(newValue) => setSearchValue(newValue.target.value)}
+                onKeyDown={(event) => {
+                  if (
+                    selectedItem.length &&
```
Comment (wei2912) [inline src/components/search-bar.jsx:116]: Can we use `> 0`?

File: src/components/search-bar.jsx [hunk 7]
```diff
@@ -67,13 +69,98 @@ const SearchBar = ({ mentors, setHasSearch, setSearchResults }) => {
...
+                    !inputValue.length &&
```
Comment (wei2912) [inline src/components/search-bar.jsx:117]: See L116.

File: src/components/search-bar.jsx [hunk 8]
```diff
@@ -67,13 +69,98 @@ const SearchBar = ({ mentors, setHasSearch, setSearchResults }) => {
...
+                }}
+                value={searchValue}
+              />
+              {isOpen ? (
+                <div style={{ border: "1px solid #ccc", position: "absolute" }}>
+                  {suggestions
+                    .filter(
+                      (i) =>
+                        !inputValue ||
+                        i.toLowerCase().includes(inputValue.toLowerCase())
```
Comment (wei2912) [inline src/components/search-bar.jsx:133]: Let's use `startsWith` instead.

File: src/components/search-bar.jsx [hunk 9]
```diff
@@ -67,13 +69,98 @@ const SearchBar = ({ mentors, setHasSearch, setSearchResults }) => {
...
+                    )
+                    .slice(0, 10)
```
Comment (wei2912) [inline src/components/search-bar.jsx:135]: Remove this, implement scrolling in place.

File: src/components/search-bar.jsx [hunk 10]
```diff
@@ -67,13 +69,98 @@ const SearchBar = ({ mentors, setHasSearch, setSearchResults }) => {
...
+                    );
+                  }
+                }}
+                value={searchValue}
+              />
+              {isOpen ? (
+                <div style={{ border: "1px solid #ccc", position: "absolute" }}>
+                  {suggestions
+                    .filter(
+                      (i) =>
```
Comment (wei2912) [inline src/components/search-bar.jsx:131]: Please rename `i` to `industry`.

File: src/components/search-bar.jsx [hunk 11]
```diff
@@ -67,13 +69,98 @@ const SearchBar = ({ mentors, setHasSearch, setSearchResults }) => {
...
+                        !inputValue ||
+                        i.toLowerCase().includes(inputValue.toLowerCase())
+                    )
+                    .slice(0, 10)
+                    .map((item, index) => {
```
Comment (wei2912) [inline src/components/search-bar.jsx:136]: Please change `index` to `i`.

**Review by wei2912 — CHANGES_REQUESTED** [review]
Please incorporate some styling with scroll bar, thanks!

File: src/components/search-bar.jsx [hunk 12]
```diff
@@ -67,13 +69,98 @@ const SearchBar = ({ mentors, setHasSearch, setSearchResults }) => {
...
+                    selectedItem.length &&
+                    !inputValue.length &&
+                    event.key === "Backspace"
+                  ) {
+                    setSelectedItem(
+                      selectedItem.slice(0, selectedItem.length - 1)
+                    );
+                  }
+                }}
+                value={searchValue}
```
Comment (bnjmnt4n) [inline src/components/search-bar.jsx:125]: I believe that removing the `onChange` and `onKeydown` should work fine if we replace `searchValue` here with `inputValue`.

**Pull request merged**